src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -184,10 +184,12 @@ trait SamRecord {
   @inline final def readGroup: SAMReadGroupRecord = getReadGroup
 
   // Use apply/update for tag attributes
-  @inline final def apply[A](name: String): A              = getAttribute(name).asInstanceOf[A]
-  @inline final def get[A](name: String): Option[A]        = Option(apply(name))
-  @inline final def update(name: String, value: Any): Unit = setAttribute(name, value)
-  @inline final def attributes: Map[String,Any] = getAttributes.map(x => x.tag -> x.value).toMap
+  @inline final def apply[A](name: String): A                 = getAttribute(name).asInstanceOf[A]
+  @inline final def get[A](name: String): Option[A]           = Option(apply(name))
+  @inline final def update(name: String, value: Any): Unit    = setAttribute(name, value)
+  @inline final def attributes: Map[String,Any]               = getAttributes.map(x => x.tag -> x.value).toMap
+  @inline final def getOrElse[A](name: String, default: A): A = get(name).getOrElse(default)
+  @inline final def contains(name: String): Boolean           = hasAttribute(name)
 
```

Comment:
While you're in here how about adding:
`@inline final def remove(name: String): Unit = setAttribute(name, null)`
to hide that ugliness from users of the API?

Reply:
I debated having a static variable named `val NoTag: String = null` but that seemed lame.